modules/cudf/src/series/string.ts

```diff
@@ -25,15 +26,25 @@ export class StringSeries extends Series<Utf8String> {
   /**
    * Casts the values to a new dtype (similar to `static_cast` in C++).
    *
-   * @param dataType The new dtype.
+   * @param type The new dtype.
    * @param memoryResource The optional MemoryResource used to allocate the result Series's device
    *   memory.
    * @returns Series of same size as the current Series containing result of the `cast` operation.
    */
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  cast<R extends DataType>(dataType: R, _memoryResource?: MemoryResource): Series<R> {
-    throw new Error(`Cast from ${arrow.Type[this.type.typeId]} to ${
-      arrow.Type[dataType.typeId]} not implemented`);
+  cast<R extends DataType>(type: R, memoryResource?: MemoryResource): Series<R> {
+    if (this.type.compareTo(type)) { return Series.new<R>(this._col as Column<R>); }
+    if (arrow.DataType.isDictionary(type)) {
+      const vals = this.cast(type.dictionary).unique(true, memoryResource);
+      const keys = this.encodeLabels(vals, undefined, undefined, memoryResource);
+      return Series.new<R>(new Column({
+        type: new Categorical(type.dictionary) as R,
+        length: keys.length,
+        nullMask: this.mask,
+        children: [keys._col, vals._col]
+      }));
+    }
```

Comment:
Refactoring this casting logic is planned in a future PR. The idea is the top-level `Series` will have a `cast()` method, which will check the dtype and call an appropriate protected `_castTo<Type>()` function, which each Series specialization can override:

```typescript
class Series {
  public cast(dataType, memoryResource?) {
    switch (dataType.typeId) {
      case arrow.Type.Integer: return this._castToInt(dataType, memoryResource);
      // etc.
    }
  }
  protected castToInt() { throw new Error("unimplemented"); }
  // etc.
}

class StringSeries extends Series {
  protected castToInt(dataType, memoryResource) {
    return Series.new(this._col.stoi(dataType, memoryResource));
  }
}
```